test/plugin/base_test.rb

```diff
@@ -626,6 +630,87 @@ def test_struct_payload_json_container_log
     end
   end
 
+  # Verify that invalid tags are sanitized. Detailed implementation differs
+  # between the grpc and non-gprc paths, which is reflected in
+  # 'setup_logging_stubs_block' and 'verify_log_name_block'.
+  # When 'require_valid_tags' is true, we only accept string tags with valid
+  # characters.
+  def test_sanitize_invalid_tags_with_requires_valid_tags_true
+    setup_gce_metadata_stubs
+    {
+      'test' => 'test',
+      'germanß' => 'german%C3%9F',
+      'chinese中' => 'chinese%E4%B8%AD',
+      'specialCharacter/_-.' => 'specialCharacter%2F_-.'
+    }.each do |tag, converted_tag|
+      setup_logging_stubs(COMPUTE_PARAMS.merge(log_name: converted_tag)) do
+        @logs_sent = []
+        d = create_driver(REQUIRE_VALID_TAGS_CONFIG, tag)
+        d.emit('msg' => log_entry(0))
```

Comment:
Added tests `test_encode_tags_from_container_name_with_require_valid_tags_true` and `test_sanitize_tags_from_container_name_with_require_valid_tags_false`.